python/lsst/pipe/supertask/parser.py

```diff
@@ -0,0 +1,278 @@
+#
+# LSST Data Management System
+# Copyright 2017 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+"""
+Module defining CmdLineParser class and related methods.
+"""
+
+#--------------------------------
+#  Imports of standard modules --
+#--------------------------------
+from argparse import Action, ArgumentParser, RawDescriptionHelpFormatter
+import collections
+import textwrap
+
+#-----------------------------
+# Imports for other modules --
+#-----------------------------
+import lsst.pipe.base.argumentParser as base_parser
+
+#----------------------------------
+# Local non-exported definitions --
+#----------------------------------
+
+__all__ = ['makeParser']
+
+# Class which keeps a single override, each -c and -C option is converted into
+# instance of this class and are put in a single ordered collection so that
+# overrides can be applied later.
+_Override = collections.namedtuple("_Override", "type,value")
+
+
+class _LogLevelAction(Action):
+    """Action class which collects logging levels.
+    """
```

Comment:
These two classes are implementation of the `argparse.Action` class and do not add any new arguments and don't define constructors. I'll extend their description and add a reference to `argparse.Action` class but I do not like describing all arguments from the base class method (`argparse` module already documents those).